src/authentication.ts

```diff
@@ -168,4 +212,9 @@ export class OAuthAuthentication implements Authentication {
                 });
         });
     }
+
+    // eslint-disable-next-line require-jsdoc
+    private isCertificateAuthentication(config: OAuthConfig): boolean {
+        return !config.password && config.certificate != null && config.privateKey != null;
```

Comment:
Should be changed to the following:
```suggestion
        return !config.password && !!config.certificate && !!config.privateKey;
```
The current behaviour will return true in case of empty string for certificate/privateKey.
Add some tests to verify what will happen if certificate/privateKey is empty or not passed (maybe add some verification that they are not empty strings?)